### PR TITLE
Supports infinite symbolic diff for gelu

### PR DESCRIPTION
**Pull request by yueyericardo**
As per title

gelu_backward code taken from `torch/_decomp/decompositions.py`

https://github.com/pytorch/pytorch/blob/80231d0a72453573728242daa0dbc9cc7b45669c/torch/_decomp/decompositions.py#L199-L226

### CODE REVIEW

**Comment by davidberard98** [comment]
@yueyericardo Thanks for the PR. Can you explain the motivation for this change? (i.e. why do we prefer writing out the decomposition instead of calling gelu_backward directly?)

**Comment by yueyericardo** [comment]
@davidberard98 Thanks for the quick reply!
The benefit of writing decomposition in `symbolic_script.cpp` is that it could generate **fused** high-order derivative graph.

However, if we use `gelu_backward` directly and we would like to compute the higher order derivatives, autodiff will fall back to using autograd for the node because there isn't a gradient formula for it.  In this case, gelu backward will use this implementation:

https://github.com/pytorch/pytorch/blob/2c0b11b43bc370d05e7899e152e01b308b42cd2f/aten/src/ATen/native/Activation.cpp#L737-L744

And it will not be fused and create much more kernels for derivatives.

**Comment by yueyericardo** [comment]
Thanks for the help from @zasdfgbnm on debugging the failed unittest.

@jjsjann123, The failed test here is from:
https://github.com/pytorch/pytorch/blob/2c0b11b43bc370d05e7899e152e01b308b42cd2f/torch/csrc/jit/codegen/cuda/type_inference.cpp#L33

And it could fixed by: https://github.com/csarofeen/pytorch/pull/1848, could you help cherry-pick this commit into upstream?

**Comment by yueyericardo** [comment]
@davidberard98 @jjsjann123 
BTW, I have a similar PR that adding support for `SiLU`. Could you also take a look of this https://github.com/pytorch/pytorch/pull/81724?

**Comment by jjsjann123** [comment]
> Thanks for the help from @zasdfgbnm on debugging the failed unittest.
> 
> @jjsjann123, The failed test here is from:
> 
> https://github.com/pytorch/pytorch/blob/2c0b11b43bc370d05e7899e152e01b308b42cd2f/torch/csrc/jit/codegen/cuda/type_inference.cpp#L33
> 
> And it could fixed by: [csarofeen#1848](https://github.com/csarofeen/pytorch/pull/1848), could you help cherry-pick this commit into upstream?

Sure thing. will push that to upstream directly.

File: torch/csrc/jit/runtime/symbolic_script.cpp [hunk 1]
```diff
@@ -916,7 +916,29 @@ const std::vector<std::string> functions = {
         def gelu(self : Tensor, *, approximate : str):
             result = torch.gelu(self, approximate=approximate)
             def backward(grad_output):
-                return torch.gelu_backward(grad_output, self, approximate=approximate), None
+                M_SQRT2 = 1.41421356237309504880
+                M_SQRT1_2 = 0.70710678118654752440
+                M_2_SQRTPI = 1.12837916709551257390
+                if approximate == "tanh":
```
Comment (jjsjann123) [inline torch/csrc/jit/runtime/symbolic_script.cpp:922]: I get nervous whenever I see control flow. In the case for approximate, `torch.nn.modules.GELU` has approximate annotated as `__constant__` so that should be fine.
But using gelu api directly with approximate as an input would stop fusion of gelu to things outside of the block. Fusing gelu_backward with neighboring ops is important.

I know this is harder, but could we instead inject the infinitely differentiable primitive implementation at `gelu_backward` instead? Basically adding an entry here for `gelu_backward` while keeping the entry for `gelu` unchanged.

cc'ing @kevinstephano

**Comment by yueyericardo** [comment]
@jjsjann123 Thanks for the quick reply!

File: torch/csrc/jit/runtime/symbolic_script.cpp (hunk 1 above)
Comment (yueyericardo) [reply torch/csrc/jit/runtime/symbolic_script.cpp:922]: Let me know if I understand correctly. So I need an entry here for gelu_backward and also define the gelu_double_backward?

I found an implementation of gelu_double_backward from 

https://github.com/pytorch/pytorch/blob/84c8a9f88e15b6d94407e8d1839db1ea8eb6ac6a/torch/csrc/autograd/FunctionsManual.cpp#L3026-L3030

And adding it to `symbolic_script.cpp`: https://github.com/yueyericardo/pytorch/commit/88a6b37e43265e745c95e811ed3f893320ddb529
But I found that 
1. If I call `torch.autograd.grad()` to calculate derivatives: it call `GeluBackwardCUDAKernelImpl` for derivatives instead of the fused kernel.
2. If I run `out.backward()`: it will match to the gelu_backward schema in `torch/csrc/jit/codegen/cuda/parser.cpp`, instead of the entry define at `symbolic_script.cpp`

https://github.com/pytorch/pytorch/blob/8d753c806275fb3436600749aaec87575dd0fabb/torch/csrc/jit/codegen/cuda/parser.cpp#L4095-L4098

**Comment by jjsjann123** [comment]
Started the cherry-pick to unblock your PR. #81792
